doc/ref/string.xml

```diff
@@ -227,17 +227,30 @@ such digits. The meaning is given in the following list
 <Index>octal character codes</Index>
 <C>\XYZ</C></Mark>
 <Item>
-    with <C>X</C>, <C>Y</C>, <C>Z</C> three octal digits.
+    with <C>X</C>, <C>Y</C>, <C>Z</C> three octal digits, that is one
+    of <C>"01234567"</C>.
     This is translated to the character corresponding to the number
     <C>X * 64 + Y * 8 + Z modulo 256</C>.
     This can be used to specify and store arbitrary binary data as a string
     in &GAP;.
 </Item>
 <Mark>
+<Index><C>\0xYZ</C></Index>
+<Index>hexadecimal character codes</Index>
+<C>\xYZ</C></Mark>
+<Item>
+    with <C>Y</C>, and <C>Z</C> hexadecimal digits, that is one of
+    <C>"0123456789ABCDEFabcdef"</C>, where <C>a</C> to <C>f</C> and
+    <C>A</C> to <C>F</C> are interpreted as the numbers <C>10</C> to
+    <C>15</C>. This is translated to the character corresponding
+    to the number <C>Y*16 + Z</C>.
+</Item>
+<Mark>
 <Index>escaping non-special characters</Index>
 other</Mark>
 <Item>
-    For any other character the backslash is simply ignored.
+  For any other character the backslash is ignored. If the character
+  is a letter, that is one of <C>a..zA..Z</C>, then a warning is displayed.
```

Comment:
Same as above: If we can't agree on changing the documented and actual behaviour, then this last sentence is controversial. In that case, we should drop it, but can keep the rest of this change.